src/libraries/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Certificate2Collection.cs

```diff
@@ -229,5 +230,94 @@ public void RemoveRange(X509Certificate2Collection certificates)
+        /// </para>
+        /// <para>
+        /// More advanced scenarios for loading certificates and
+        /// can leverage <see cref="System.Security.Cryptography.PemEncoding" /> to enumerate
+        /// PEM-encoded values and apply any custom loading behavior.
+        /// </para>
+        /// </remarks>
+        /// <exception cref="CryptographicException">
+        /// The decoded contents of a PEM are invalid or corrupt and could not be imported.
+        /// </exception>
+        /// <exception cref="ArgumentNullException">
+        /// <pararef name="certPemFilePath" /> is null.
+        /// </exception>
+        public void ImportFromPemFile(string certPemFilePath)
+        {
+            if (certPemFilePath is null)
+                throw new ArgumentNullException(nameof(certPemFilePath));
+
+            ReadOnlySpan<char> contents = System.IO.File.ReadAllText(certPemFilePath);
+            ImportFromPem(contents);
+        }
+
+        /// <summary>
+        /// Imports a collection of RFC 7468 PEM-encoded certificates.
+        /// </summary>
+        /// <param name="certPem">The text of the PEM-encoded X509 certificate collection.</param>
+        /// <remarks>
+        /// <para>
+        /// PEM-encoded items with a CERTIFICATE PEM label will be imported. PEM items
+        /// with other labels will be ignored.
+        /// </para>
+        /// <para>
+        /// More advanced scenarios for loading certificates and
+        /// can leverage <see cref="System.Security.Cryptography.PemEncoding" /> to enumerate
+        /// PEM-encoded values and apply any custom loading behavior.
+        /// </para>
+        /// </remarks>
+        /// <exception cref="CryptographicException">
+        /// The decoded contents of a PEM are invalid or corrupt and could not be imported.
+        /// </exception>
+        public void ImportFromPem(ReadOnlySpan<char> certPem)
+        {
+            int added = 0;
+
+            try
+            {
+                foreach ((ReadOnlySpan<char> contents, PemFields fields) in new PemEnumerator(certPem))
+                {
+                    ReadOnlySpan<char> label = contents[fields.Label];
+
+                    if (label.SequenceEqual(PemLabels.X509Certificate))
+                    {
+                        // We verify below that every byte is written to.
+                        byte[] certBytes = GC.AllocateUninitializedArray<byte>(fields.DecodedDataLength);
+
+                        if (!Convert.TryFromBase64Chars(contents[fields.Base64Data], certBytes, out int bytesWritten)
+                            || bytesWritten != fields.DecodedDataLength)
+                        {
+                            // The contents should have already been validated by the PEM reader.
+                            throw new ArgumentException(SR.Cryptography_X509_NoPemCertificate, nameof(certPem));
+                        }
+
+                        Import(certBytes);
```

Comment:
Similar to the comment for single-certs, this would support loading PFX contents with the CERTIFICATE PEM label.  That seems weird, and this should probably use the aforementioned new single-DER pal method then just call Add.